src/DIRAC/Resources/Computing/HTCondorCEComputingElement.py

```diff
@@ -152,11 +108,49 @@ def __init__(self, ceUniqueID):
         self.tokenFile = None
 
     #############################################################################
-    def __writeSub(self, executable, nJobs, location, processors, pilotStamps, tokenFile=None):
+
+    def _DiracToCondorID(self, diracJobID):
+        """Convert a DIRAC jobID into an Condor jobID.
```

Comment:
```suggestion
        """Convert a DIRAC jobID into an Condor jobID.
```
The use of DIRAC jobID is confusing here, this has nothing to do with a DIRAC job, but rather a pilot identifier?
So it turns a pilot Identifier into a condor jobID

Reply:
You are right, the name is not correct but here is an explanation. I wanted to bring more homogeneity between the `AREX` and the `HTCondor` CE interfaces. We use the `AREXCE` to submit pilot-jobs but also jobs. Theoretically, we could also use the `HTCondorCE` to submit jobs, which is why I did not want to call this a pilot identifier.

But indeed, it is neither a dirac job ID nor a dirac pilot reference. I would like to find a generic name for that (`genericJobID`? `jobID` - as opposed to `condorJobID`? `jobReference`?). Any opinion? Suggestion?

Reply:
> We use the AREXCE to submit pilot-jobs but also jobs.

I had no idea about this possibility. What does this thing look like in case this is a "job"?

Reply:
> I had no idea about this possibility. What does this thing look like in case this is a "job"?

We use the `AREXCE` to push an application (e.g. a bash script containing `<command> <arguments>`) from a workflow module and wait for its termination, as part of the `PushJobAgent` execution (to execute jobs in computing resources with no external connectivity).
In the future, we would like to directly push the `dirac-jobexec` command using the `AREXCE`.

Reply:
Thanks for the explanation!
What does `diracJobID` look like in that case?

Reply:
There is no difference, the `diracJobID` is the same in both cases, namely a transformation of the CE-specific ID to a "generic" format: `1234` becomes `<protocol>://<cename + extra fields like port>/1234`.
Then:
- if it is a pilot-job, it is interpreted by the caller of the CE (the `SiteDirector`) as a pilot job reference.
- if it is an application, it is interpreted by the caller of the CE (a Workflow module) as a "job-application" ID let's say (it is just named `jobID` in the code, you can see that in `RemoteRunner`).

Does this answer to your question?

Reply:
Yeah, so calling this a jobReference is probably best?